src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -0,0 +1,275 @@
+"""Interface to the GPU NUFFT library."""
+
+import numpy as np
+from .base import FourierOperatorBase, proper_trajectory
+
+GPUNUFFT_AVAILABLE = True
+try:
+    from gpuNUFFT import NUFFTOp
+except ImportError:
+    GPUNUFFT_AVAILABLE = False
+
+
+class gpuNUFFT:
+    """GPU implementation of N-D non uniform Fast Fourrier Transform class.
+
+    Attributes
+    ----------
+    samples: np.ndarray
+        the normalized kspace location values in the Fourier domain.
+    shape: tuple of int
+        shape of the image
+    operator: The NUFFTOp object
+        to carry out operation
+    n_coils: int default 1
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition. If n_coils > 1, please organize data as
+            n_coils X data_per_coil
+    """
+
+    def __init__(
+        self,
+        samples,
+        shape,
+        n_coils=1,
+        density_comp=None,
+        kernel_width=3,
+        sector_width=8,
+        osf=2,
+        balance_workload=True,
+        smaps=None,
+    ):
+        """Initilize the 'NUFFT' class.
+
+        Parameters
+        ----------
+        samples: np.ndarray
+            the kspace sample locations in the Fourier domain,
+            normalized between -0.5 and 0.5
+        shape: tuple of int
+            shape of the image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition
+        density_comp: np.ndarray default None.
+            k-space weighting, density compensation, if not specified
+            equal weightage is given.
+        kernel_width: int default 3
+            interpolation kernel width (usually 3 to 7)
+        sector_width: int default 8
+            sector width to use
+        osf: int default 2
+            oversampling factor (usually between 1 and 2)
+        balance_workload: bool default True
+            whether the workloads need to be balanced
+        smaps: np.ndarray default None
+            Holds the sensitivity maps for SENSE reconstruction
+        """
+        if GPUNUFFT_AVAILABLE is False:
+            raise ValueError(
+                "gpuNUFFT library is not installed, " "please refer to README"
```

Comment:
Why two strings ?
I didn't even know it was syntactically accepted to write two strings in a row